### PR TITLE
Add support for invisible reCAPTCHAs

### LINKED ISSUE #7674: Invisible reCaptcha

### PR DESCRIPTION
**Pull request by jimchamp**
Closes #7674

Replaces our image-based reCAPTCHA with an invisible reCAPTCHA.  ~Not yet tested.~  

**Edit:** I was able to create an account on `testing`.

### CODE REVIEW

**Comment by cdrini** [comment]
Tried to create an account on testing ; the invisible captcha floating button thin gappeared, but clicking "sign up" caused the button to become disabled for a bit, and then it became re-enabled? Didn't receive any email.

File: openlibrary/templates/account/create.html [hunk 1]
```diff
@@ -57,21 +57,16 @@ <h1>$_("Sign Up")</h1>
 
         <hr/>
 
-        $if form.has_recaptcha:
-            <div class="formElement">
-                <div class="label smaller lighter">$_("If you have security settings or privacy blockers installed, please disable them to see the reCAPTCHA.")</div>
-                <div class="g-recaptcha" data-sitekey="$form['recaptcha'].public_key"></div>
```
Comment (cdrini) [inline openlibrary/templates/account/create.html:63]: Do we need to update all other uses of this? `g-recaptcha` is used in a few places

**Comment by mekarpeles** [comment]
This code looks like it work! One weird edge case with aliases which I think we fixed:

```
diff --git a/openlibrary/accounts/model.py b/openlibrary/accounts/model.py
index 7f0126942..c3862e244 100644
--- a/openlibrary/accounts/model.py
+++ b/openlibrary/accounts/model.py
@@ -657,20 +657,19 @@ class InternetArchiveAccount(web.storage):
                 service='openlibrary',
             )
 
+            vs = response.get('values', {})
+            vsj = ' '.join(vs.values())
             if response.get('success'):
                 ia_account = cls.get(email=email)
                 if test:
                     ia_account.test = True
                 return ia_account
-
-            elif 'screenname' not in response.get('values', {}):
-                errors = '_'.join(response.get('values', {}))
-                raise ValueError(errors)
-
+            # If we have errors besides taken screenname, abort
+            elif list(vs) != ['screenname']:
+                raise ValueError(vsj)
+            # If we're out of retries
             elif attempt >= retries:
-                ve = ValueError('username_registered')
-                ve.value = _screenname
-                raise ve
+                raise ValueError(vsj)
 
             _screenname = append_random_suffix(screenname)
             attempt += 1
diff --git a/openlibrary/plugins/upstream/account.py b/openlibrary/plugins/upstream/account.py
index c8956cde2..2bd25f049 100644
--- a/openlibrary/plugins/upstream/account.py
+++ b/openlibrary/plugins/upstream/account.py
@@ -283,7 +283,9 @@ class account_create(delegate.page):
                 """  # nopep8
                 mls = ['ml_best_of', 'ml_updates']
                 notifications = mls if f.ia_newsletter.checked else []
-                InternetArchiveAccount.create(
+                email = f.email
+                email_val = email.value
+                r = InternetArchiveAccount.create(
                     screenname=f.username.value,
                     email=f.email.value,
                     password=f.password.value,
@@ -294,8 +296,8 @@ class account_create(delegate.page):
                 return render['account/verify'](
                     username=f.username.value, email=f.email.value
                 )
-            except ValueError:
-                f.note = LOGIN_ERRORS['max_retries_exceeded']
+            except ValueError as ve:
+                f.note = str(ve)
 
         return render['account/create'](f)
```

File: openlibrary/templates/account/create.html (hunk 1 above)
Comment (mekarpeles) [reply openlibrary/templates/account/create.html:63]: Tested, I think it's good as is!

**Comment by mekarpeles** [comment]
Code tested and works, there are some challenges with alias accounts i.e. `mek+blah@archive.org` but I believe that can be handled in a separate PR which addresses https://github.com/internetarchive/openlibrary/pull/8258#issuecomment-1804740777

**Pull request merged**